sql/timeseries/h3.sql

```diff
@@ -0,0 +1,32 @@
+#standardSQL
+# The amount of requests either using HTTP/3 or able to use it.
+#
+# We measure "ability to use" as well as "actual use", as HTTP Archive is a
+# cold crawl and so less likely to use HTTP/3 which requires prior visits.
+# 
+# For "able to use" we look at the alt-svc response header.
+#
+# We also only measure official HTTP/3 (ALPN h3, h3-29) and not gQUIC or other
+# prior versions. h3-29 is the final draft version and will be switched to h3
+# when HTTP/3 is approved so we include that as it is HTTP/3 in all but name.
+#
+SELECT
+  SUBSTR(_TABLE_SUFFIX, 0, 10) AS date,
+  UNIX_DATE(CAST(REPLACE(SUBSTR(_TABLE_SUFFIX, 0, 10), '_', '-') AS DATE)) * 1000 * 60 * 60 * 24 AS timestamp,
+  IF(ENDS_WITH(_TABLE_SUFFIX, 'desktop'), 'desktop', 'mobile') AS client,
+  ROUND(SUM(IF(respHttpVersion = 'HTTP/3' OR respHttpVersion = 'h3' OR respHttpVersion = 'h3-29'
+                OR reqHttpVersion = 'HTTP/3' OR reqHttpVersion = 'h3' OR reqHttpVersion = 'h3-29'
+                OR REGEXP_EXTRACT(REGEXP_EXTRACT(respOtherHeaders, r'alt-svc = (.*)'), r'(.*?)(?:, [^ ]* = .*)?$') LIKE '%h3=%'
+                OR REGEXP_EXTRACT(REGEXP_EXTRACT(respOtherHeaders, r'alt-svc = (.*)'), r'(.*?)(?:, [^ ]* = .*)?$') LIKE '%h3-29=%'
+              , 1, 0)) * 100 / COUNT(0), 2) AS percent
+FROM
+  `httparchive.summary_requests.*`
+WHERE
+  SUBSTR(_TABLE_SUFFIX, 0, 10) >= '2020_01_01'
```

Comment:
I wonder if for query performance purposes it'd be faster to bake the decade into the table and wildcard the rest, for example:
```suggestion
  `httparchive.summary_requests.202*`
```

It'd also require updating the date/timestamp fields above. And yes this won't handle 2030+ but I'd be shocked if we're not using something else by then 😁

Reply:
Seems to be same speed/data queried (3TB each time).

Note that changing this to `SUBSTR(_TABLE_SUFFIX, 0, 10) >= '2018_01_01'` makes it 4.8 TB so it's definitely having the desired effect.